models/discrete/populations.py

```diff
@@ -285,59 +374,77 @@ def compile(self):
 class Indexed_Population(Population):
     def __init__(self, name: str, shape: tuple):
         super().__init__(name)
-        self.array = np.empty(shape, dtype = object)
+        self.array = np.empty(shape, dtype=object)
 
     def __getitem__(self, index):
         return self.array[index]
-    
+
     def add_population(self, population, index):
         super()._add_population(population)
         self.array[index] = population
-        
-
 
 
-'''
+"""
 Window classes
-'''
+"""
 
 
 class Window:
-    def __init__(self, name: str, variables: Variables = Variables(), operators: Operators = Operators()):
+    def __init__(
+        self,
+        name: str,
+        variables: Variables = Variables(),
+        operators: Operators = Operators(),
+    ):
         self.name = name
         self.variables = variables
         self.operators = operators
 
-    def _add_operator(self, name, function, description = None):
-        operator = Operator(name, function, description or f"{self.name} operator: {name}")
+    def _add_operator(self, name, function, description=None):
+        operator = Operator(
+            name, function, description or f"{self.name} operator: {name}"
+        )
         self.operators += operator
         setattr(self, name, operator)
 
 
-'''
+"""
 System
-'''
+"""
+
 
 class PopulationSystemError(Exception):
     pass
 
-class System:
+
+
+
+
+class System():
     def __init__(self, population):
         self.population = population
-        self.system_time = Variable(T, [0], "system time")
-        self.system_variables = Variables([self.system_time])
-        self.variables = self.population.variables + self.system_variables
-        self.parameters = self.population.parameters
+        self.variables = Variables([time])
+        self.parameters = Parameters()
         self.update_rules = self.population.update_rules._combine_update_rules()
```

Comment:
These update rules should become part of the SimVariables in some shape or form (I see you put an equation attribute there, maybe for now that's good enough). So we don't need to store them as a separate attribute.

Reply:
This should be dealt with now, the combined update rule for each variable is computed and stored in `variable.equation`.